app/src/main/cpp/skyline/services/nvdrv/INvDrvServices.cpp

```diff
@@ -1,149 +1,134 @@
-// SPDX-License-Identifier: MPL-2.0
+// SPDX-License-Identifier: MIT OR MPL-2.0
 // Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
 
 #include <kernel/types/KProcess.h>
 #include "INvDrvServices.h"
 #include "driver.h"
 #include "devices/nvdevice.h"
 
+#define NVRESULT(x) [&response, this](NvResult err) {     \
+        if (err != NvResult::Success)                     \
+            state.logger->Debug("IOCTL Failed: {}", err); \
+                                                          \
+        response.Push<NvResult>(err);                     \
+        return Result{};                                  \
+    } (x)
+
 namespace skyline::service::nvdrv {
-    INvDrvServices::INvDrvServices(const DeviceState &state, ServiceManager &manager) : driver(nvdrv::driver.expired() ? std::make_shared<Driver>(state) : nvdrv::driver.lock()), BaseService(state, manager) {
-        if (nvdrv::driver.expired())
-            nvdrv::driver = driver;
-    }
+    INvDrvServices::INvDrvServices(const DeviceState &state, ServiceManager &manager, Driver &driver, const SessionPermissions &perms) : BaseService(state, manager), driver(driver), ctx(SessionContext{.perms = perms}) {}
 
     Result INvDrvServices::Open(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
-        auto path{request.inputBuf.at(0).as_string()};
+        constexpr FileDescriptor SessionFdLimit{sizeof(u64) * 2 * 8}; //!< Nvdrv uses two 64 bit variables to store a bitset
```

Comment:
So smth like 

template <typename T>
using BitsPer = std::numeric_limits<T>::digits
In common? Or would you rather me use numeric limits directly with u8 here

Reply:
I guess considering the fact you can eliminate the `sizeof(u64)` by inlining it and we basically use `BitsPerByte` nowhere, it might be be better to use it directly.